src/Corax/IndexSearcher/IndexSearcher.cs

```diff
@@ -173,7 +173,9 @@ internal ByteStringContext<ByteStringMemoryCache>.InternalScope ApplyAnalyzer(Sl
     [MethodImpl(MethodImplOptions.AggressiveInlining)]
     private ByteStringContext<ByteStringMemoryCache>.InternalScope AnalyzeTerm(IndexFieldBinding binding, ReadOnlySpan<byte> originalTerm, int fieldId, out Slice value)
     {
-        var analyzer = binding.Analyzer!;
+        var analyzer = binding.FieldIndexingMode is FieldIndexingMode.Search 
+            ? Analyzer.DefaultLowercaseAnalyzer // lowercase only when search is used in non-full-text-search match 
```

Comment:
Shouldn't we always have an analyzer for the `binding`, and then we won't need it?

Reply:
In this place, we always have an analyzer. This is a case when we have a search analyzer assigned for the field. We don't want to use it when performing queries differently than `search()`.